### PR TITLE
Various project build file improvements

### PR DESCRIPTION
**Pull request by hzsweers**
Standardizes a few declarations, uses apt-idea for better IDE support, and updates remaining project plugins to the new `plugins` DSL

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -14,6 +14,41 @@
  * limitations under the License.
  */
 
+pluginManagement {
+  resolutionStrategy {
+    eachPlugin {
+      switch (requested.id.id) {
+        case 'org.jetbrains.dokka':
+          useModule('org.jetbrains.dokka:dokka-gradle-plugin:0.9.16-eap-3')
```
Comment (kageiit) [inline settings.gradle]: might be good to move these versions to deps.gradle?

Its also atypical to put this in settings.gradle. Cant it be in the root build.gradle at the top?

Comment (hzsweers) [reply settings.gradle]: Can't apply those before settings.gradle. Want to get a baseline done first

And this has to be done in settings.gradle per the documentation - https://docs.gradle.org/current/userguide/plugins.html#sec:custom_plugin_repositories

Comment (hzsweers) [reply settings.gradle]: If we make a buildSrc, we could do that. I'll probably just move these versions to be consolidated here though for now

Comment (kageiit) [reply settings.gradle]: im surprised we cannot `apply from: 'deps.gradle'` in settings.gradle. Looks fine then

Comment (hzsweers) [reply settings.gradle]: Actually let me try, I think I tried it on a different project with a more complex dependencies file than this one

Comment (hzsweers) [reply settings.gradle]: yay it worked - 8091195

Comment (kageiit) [reply settings.gradle]: +1

**Pull request merged**